recorder/src/main/java/fk/prof/PerfCtx.java

```diff
@@ -0,0 +1,95 @@
+package fk.prof;
+
+/**
+ * @understands a named "logical" context under which all performance data is aggregated
+ * <p>
+ * Note about constructors:
+ * Creates a perf-data-recording-context with desired merge semantics. This internally checks if a conflicting duplicate context (same name, but different properties) are created with conflicting coverage-pct and/or merge-semantics) creation is being attempted and fails the call.
+ * <p>
+ * So the best way to use this is to create a static+final instance and use it in the relevant places. If that is not possible, separate instances may be used, but user must take care to not provide conflicting values for these parameters. Also, constructor here has non-trivial performance-cost, so its best used as static+final instance.
+ * <p>
+ * Constraints / Limits:
+ * 1. An application (single JVM instance) can have only upto 100 unique PerfCtx instances (uniqueness is implied by name)
+ * 2. There exists an upper-limit to number of context to be tracked. This may change between releases. Consult the package-maintainers to learn the actual enforced limit.
+ * 3. No more than 6 perf-ctx instances should be nested with "scoped" merge-semantic.
```

Comment:
Yes, I choose to give people more ctxs at the cost of lesser nesting depth.

The other way, trade-off wasn't working great.

As in, to bring it  down to 6, we had to give up a lot in number of user-defined ctx. It falls to 118, which I thought was too low (keeping in mind that people would use libraries etc).